zulipterminal/ui_tools/boxes.py

```diff
@@ -810,6 +814,40 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
         elif is_command_key("MARKDOWN_HELP", key):
             self.view.controller.show_markdown_help()
             return key
+        elif is_command_key("OPEN_EXTERNAL_EDITOR", key):
+            editor = os.environ.get("ZULIP_EDITOR_COMMAND", os.environ.get("EDITOR"))
+            if editor is None:
+                self.view.controller.report_error(
+                    "Configure $EDITOR or $ZULIP_EDITOR_COMMAND shell environment."
+                )
+                return key
+            editor_splits = editor.split(" ")
+            if len(editor_splits) >= 1:
+                fullpath_program = shutil.which(editor_splits[0])
+                if fullpath_program is None:
+                    self.view.controller.report_error(
+                        "Editor program not found, check $EDITOR "
+                        "or $ZULIP_EDITOR_COMMAND."
+                    )
+                    return key
+                editor_splits[0] = fullpath_program
+            edit_tempfile = NamedTemporaryFile(suffix=".md")
+            with open(edit_tempfile.name, mode="w") as edit_writer:
+                edit_writer.write(self.msg_write_box.edit_text)
+            self.view.controller.loop.screen.stop()
+            if "TEMPFILE" not in editor:
+                editor_splits.append(edit_tempfile.name)
+            else:
+                editor_splits = [
+                    part if part != "TEMPFILE" else edit_tempfile.name
+                    for part in editor_splits
+                ]
```

Comment:
Is "TEMPFILE" related to a special case for `lapce`? Does appending the filename not work?

Reply:
I remove the condition and keep only appending.
I was like 'maybe some people would like to do pipe command' but it will not work with shell off.

I test it with lapce / sublime text / marker and all don't need to position the file inside the command.